datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java

```diff
@@ -39,46 +47,55 @@
 
     private static final int TIMEOUT_S = 10;
 
-    private static final String MESSAGE_ONE = "Hello here 1";
-    private static final String MESSAGE_TWO = "Hello here 2";
     private static final String FOLDER = "folder1";
     private static final String PRIVATE_FILE = "secret.txt";
 
-    private static final int NUMBER_OF_TEST_USERS = 5;//10;
-    private static final int NUMBER_OF_TEST_FILES = 10;//100;
+    private static final int NUMBER_OF_TEST_USERS = 2;//10;
+    private static final int NUMBER_OF_TEST_FILES = 5;//100;
     private static final int EXPECTED_NUMBER_OF_FILES_PER_USER = NUMBER_OF_TEST_FILES;
-    private static final int FILE_SIZE_IN_KB = 1024 * 30; //30Kb
 
     protected StorageService storage;
     protected URI location;
 
+    private TestMetricCollector<TestRecord> metricCollector = new TestMetricCollector<>();
+
     @SneakyThrows
-    @ParameterizedTest
-    @MethodSource("storages")
-    void writeToPrivateListPrivateInDifferentThreads(WithStorageProvider.StorageDescriptor descriptor) {
+    @ParameterizedTest(name = "Run #{index} service storage: {0} with data size: {1} bytes and {2} threads.")
+    @MethodSource("differentThreadsTestOptions")
+    public void writeToPrivateListPrivateInDifferentThreads(WithStorageProvider.StorageDescriptor descriptor, int size, int poolSize) {
         init(descriptor);
 
-        byte[] testData = generateTestData(FILE_SIZE_IN_KB);
+        String testFile =  "./test.txt";
+        generateTestFile(testFile, size);
+
+        ThreadPoolExecutor executor = (ThreadPoolExecutor) Executors.newFixedThreadPool(poolSize);
 
-        ThreadPoolExecutor executor = (ThreadPoolExecutor) Executors.newFixedThreadPool(4);
         CountDownLatch holdingLatch = new CountDownLatch(1);
         CountDownLatch finishHoldingLatch = new CountDownLatch(NUMBER_OF_TEST_USERS * NUMBER_OF_TEST_FILES);
 
-        String checksumOfOriginTestFile = checksum(new ByteArrayInputStream(testData));
+        String checksumOfOriginTestFile;
+        try(FileInputStream input = new FileInputStream(new File(testFile))) {
+            checksumOfOriginTestFile = checksum(input);
+        }
 
         log.trace("*** Starting write threads ***");
         for (int i = 0; i < NUMBER_OF_TEST_USERS; i++) {
             long userRegisterTime = System.currentTimeMillis();
-            UserIDAuth user = registerUser("john_" + i, location);
-            loadReport.add("Register user " + user.getUserID().getValue() + ": " +
-                    (System.currentTimeMillis() - userRegisterTime) + "ms");
-            log.debug("Registered user: {}", user.getUserID().getValue());
+            String userName = "john_" + i;
+            executor.execute(() -> {
+                UserIDAuth user = registerUser(userName, location);
```

Comment:
Having ```long userRegisterTime = System.currentTimeMillis();``` before this line in lambda body will be more precise imho and less error prone
Also for time measurement it is better to have specialized lambda function that publishes result:
```
Function<Supplier<T>, Consumer<Long>> measurePerformance = (inv, pubResult) ->
{
   Instant startSaving = Instant.now();
T result = inv.execute();
long durationOfSavingFile = Duration.between(startSaving, Instant.now()).toMillis();
pubResult.accept(durationOfSavingFile);
return result;
}

```